cmd/kubelet/app/server.go

```diff
@@ -488,13 +493,19 @@ func (s *KubeletServer) kubeconfigClientConfig() (*client.Config, error) {
 // fall back to the default auth (none) without an error.
 // TODO(roberthbailey): Remove support for --auth-path
 func (s *KubeletServer) createClientConfig() (*client.Config, error) {
-	if s.KubeConfig.Provided() && s.AuthPath.Provided() {
-		return nil, fmt.Errorf("cannot specify both --kubeconfig and --auth-path")
-	}
-	if s.KubeConfig.Provided() {
-		return s.kubeconfigClientConfig()
-	} else if s.AuthPath.Provided() {
-		return s.authPathClientConfig(false)
+	authPathFlag := pflag.Lookup(authPathFlagName)
+	kubeconfigFlag := pflag.Lookup(kubeconfigFlagName)
+	if authPathFlag != nil && kubeconfigFlag != nil {
+		if authPathFlag.Changed && kubeconfigFlag.Changed {
```

Comment:
> for the commands that will unset fields, that might make sense. @deads2k, opinion on that?

Reading this thread, I think that @liggitt  is talking about the `Options` objects from the `kubectl config` command.  For those, I'm fine with expressing the unsetting as discrete value in the struct which can be individually set.  I'm also ok with `*bool` instead `bool`.

For this method, it seems weird to have `createClientConfig` require an already existing `client.Config`, but I guess I wouldn't rule it out without seeing what it looked like.  In the current incarnation, I think it would make more sense to pass in the flagset that was actually used by the command rather than hiding the dependency behind a static access.